lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3498,6 +3499,19 @@ DynamicallyReplacedDeclRequest::evaluate(Evaluator &evaluator,
   if (attr->isInvalid())
     return nullptr;
 
+  auto *clazz = VD->getDeclContext()->getSelfClassDecl();
+  if (clazz && clazz->isDistributedActor()) {
+    // Since distributed actors synthesize their `_remote`
+    // counterparts to any `distributed func` declared on them, and such remote
+    // function is a popular target for dynamic function replacement - we must
+    // force the synthesis has happened here already, such that we can lookup
+    // the func for the replacement we're handling right now.
+    //
+    // This request is cached, so it won't cause wasted/duplicate work.
+    TypeChecker::addImplicitDistributedActorRemoteFunctions(clazz);
+  }
```

Comment:
This is the important bit (well, one of them); if we first visit an extension, and didn't yet synthesize remote functions, but the extension has an function dynamic member replacement, the function it wants to replace does not exist yet so we must trigger synthesis. We can be smarter about the synthesis later, by requesting the specific func, but realistically we need all of them anyway and this unblocks us for now.